Created documentation for Search Options feature (#244)
mozilla-l10n/localizer-documentation: src/tools/pontoon/search_filters.md

To search for an exact match, wrap the search terms in double quotes, e.g. `"new tab"`. On the other hand, to search for strings that contain double quotes, escape them with a backslash, e.g. `\"`.

> ## Options

Reviewer: (I'm OK either way, leaving the final say to @mathjazz)

> ## Options

Author: That makes more sense to me. If I add another `#` to each title doesn't seem to look terrible either.

<img width="771" alt="image" src="https://github.com/user-attachments/assets/9febc218-ea7e-4d72-94f1-ac1e2cfc16ad">

<img width="771" alt="image" src="https://github.com/user-attachments/assets/03ad8534-d352-4976-a57c-c7e3fe416b81">

In addition to the search filters listed below, Pontoon provides additional **search options**, allowing users to refine search results. Search options can be accessed by clicking the magnifying glass icon on the right side of the search box.

![Options](../../assets/images/pontoon/search_filters/options.png)

Search options include:
* **Match case**: Exactly match the capitalization of the search term.
* **Match whole word**: Provides the same result as surrounding a word with double quotes, exactly matching the search term (case insensitive), and avoiding partial matches.
* **Include string identifiers**: Expand the search to include string identifiers (keys).
* **Include rejected translations**: Expand the search to include translations that have been rejected.
* **Exclude source strings**: Avoid searching for matches in source strings.

All search options are disabled by default. Clicking anywhere on the name or checkbox will toggle only the specific search option. Once ready, clicking the `APPLY SEARCH OPTIONS` at the bottom of the panel will apply the selected search options.

![Multiple options](../../assets/images/pontoon/search_filters/options_multiple.png)

### Filters

> #### Translation status

Reviewer 2: Filters should be the same nesting level as Search.

Search options should be the same nesting level as Filter types.

In this case, there are 10 tags defined for the project (highlighted in red). Near each tag there is a representation of the priority: like for projects, it goes from 5 stars (highest priority) to 1 star (lowest priority).

Note: translation time, translation authors, and tags are not available when `All Projects` is selected as a [resource](glossary.md#resource).

> ## Options

Reviewer 2: This section should be under Search, not Filters.

Reviewer: Good point. Worth noting that the search filter will be "below" at that point (next paragraph).

> In addition to all of the search filters listed above, Pontoon also provides **search options**, allowing users to refine search results. Search options can be accessed by clicking the magnifying glass icon on the right side of the search box.

Reviewer: ```suggestion
In addition to the search filters listed above, Pontoon provides additional **search options**, allowing users to refine search results. Search options can be accessed by clicking the magnifying glass icon on the right side of the search box.
```

![Options](../../assets/images/pontoon/search_filters/options.png)

Search options include:

> * **Match case**: Exactly match the casing of the search term.

Reviewer: ```suggestion
* **Match case**: Exactly match the capitalization of the search term.
```

> * **Match whole word**: Similar to surrounding with double quotes, exactly match the search term (case insensitive).

Reviewer: ```suggestion
* **Match whole word**: Provides the same result as surrounding a word with double quotes, exactly matching the search term (case insensitive), and avoiding partial matches.
```

Not sure if this is clearer?

Author: Seems more clear to me personally!

> * **Match whole word**: Similar to surrounding with double quotes, exactly match the search term (case insensitive).

Reviewer 2: ```suggestion
* **Match whole word**: Search term must exactly match a word in the text without being part of a longer word.
```

* **Include string identifiers**: Expand the search to include string identifiers (keys).
* **Include rejected translations**: Expand the search to include translations that have been rejected.

> * **Exclude source strings**: Exclude any matches that appear within source strings.

Reviewer: ```suggestion
* **Exclude source strings**: Avoid searching for matches in source strings.
```

The current wording is potentially incorrect: we don't exclude strings that have matches in the source (but might have matches in translations).

> All search options are off by default. Clicking anywhere on the name or checkbox will toggle only this search option. Once ready, clicking the `APPLY SEARCH OPTIONS` at the bottom of the panel will apply the respective search options.

Reviewer: ```suggestion
All search options are disabled by default. Clicking anywhere on the name or checkbox will toggle only the specific search option. Once ready, clicking the `APPLY SEARCH OPTIONS` at the bottom of the panel will apply the selected search options.
```
